File: tests/kotlin/enums/test_enum_iteration.kt
// vybe-test: kotlin/enums/test_enum_iteration
// origin: languages/kotlin/tests/kotlin/test_enums.rs

enum class Light { RED, GREEN, BLUE }

        fun main() {
            var count = 0
            for (entry in arrayOf(Light.RED, Light.GREEN, Light.BLUE)) {
                if (entry == Light.RED) {
                    count += 1
                } else if (entry == Light.GREEN) {
                    count += 1
                } else {
                    count += 1
                }
            }
            println(count)
        }

File: tests/python/python_enum_advanced/test_enum_iteration.py
# vybe-test: python/python_enum_advanced/test_enum_iteration
# origin: languages/python/tests/python/test_python_enum_advanced.rs

from enum import Enum

class Day(Enum):
    MON = 1
    TUE = 2
    WED = 3

for d in Day:
    print(d.name)
